src/actions/auth.js

```diff
@@ -0,0 +1,246 @@
+
+    return {
+        type: AuthAct.REQUEST_PASSWORD_RESET_SUCCESS
+    };
+};
+
+exports.requestResetFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.REQUEST_PASSWORD_RESET_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.resetPasswordRequest = () => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_BEGIN
+    };
+};
+
+exports.resetPasswordSuccess = () => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_SUCCESS
+    };
+};
+
+exports.resetPasswordFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.requestPasswordReset = ({ email }) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.requestResetRequest());
+
+        return WebClient.post('/users/request-reset', { email }, { responseType: 'text' })
+
+        .then(({ data, status }) => {
+
+            dispatch(actions.requestResetSuccess());
+            return History.push('/login');
+
+        })
+        .catch((err) => {
+
+            let errMessage = 'Unable to reset password. Please try again';
+
+            if (typeof err.response !== 'undefined') {
+                errMessage = err.response.data.message;
+            }
+
+            dispatch(actions.requestResetFailure(errMessage));
+        });
+    };
+};
+
+exports.resetPassword = (email, newPassword, resetToken) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.resetPasswordRequest());
+
+        return WebClient.post('/users/reset-password', { email, newPassword, resetToken }, { responseType: 'text' })
+
+        .then(({ data, status }) => {
+
+            dispatch(actions.resetPasswordSuccess());
+            dispatch(actions.login({ email, password: newPassword }));
+
+        })
+        .catch((err) => {
+
+            let errMessage = 'Unable to reset password. Please try again.';
+
+            if (typeof err.response !== 'undefined') {
+                errMessage = err.response.data.message;
+            }
+
+            dispatch(actions.resetPasswordFailure(errMessage));
+        });
+    };
+};
+
+// StrangeAuth
+
+internals.strangeActions = StrangeAuth.makeActions({
+    login: ({ email, password, token }) => {
+
+        let authPromise;
+        let finalToken;
+
+        if (!!token) {
+
+            finalToken = token;
+
+            authPromise = internals.getUser(finalToken);
+
+        }
+        else {
+
+            authPromise = WebClient.post('/login', { email, password }, { responseType: 'text' })
+
+            .then(({ data, status }) => {
+
+                finalToken = data;
+
+                return internals.getUser(finalToken);
+
+            });
+        }
+
+        return authPromise.then(({ data, status }) => {
+
+            return {
+                credentials: {
+                    token: finalToken,
+                    user: data
+                }
+            };
+        });
+    }
+});
+
+internals.getUser = (token) => {
```

Comment:
The is cool—also would have been alright to define this function inside `login()`.